ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -0,0 +1,907 @@
+import com.daml.ledger.api.domain.{LedgerId, ParticipantId, PartyDetails}
+import com.daml.ledger.participant.state.index.v2.PackageDetails
+import com.daml.platform.store.Conversions.{
+  contractId,
+  eventId,
+  flatEventWitnessesColumn,
+  identifier,
+  instant,
+  ledgerString,
+  offset,
+  party,
+}
+import com.daml.ledger.participant.state.v1.{Configuration, Offset}
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.PackageId
+import com.daml.platform.store.Conversions
+import com.daml.platform.store.SimpleSqlAsVectorOf.SimpleSqlAsVectorOf
+import com.daml.platform.store.appendonlydao.JdbcLedgerDao.{acceptType, rejectType}
+import com.daml.platform.store.appendonlydao.events.{ContractId, Key}
+import com.daml.platform.store.backend.StorageBackend
+import com.daml.platform.store.backend.StorageBackend.RawTransactionEvent
+import com.daml.platform.store.entries.{ConfigurationEntry, PackageLedgerEntry, PartyLedgerEntry}
+import com.daml.platform.store.interfaces.LedgerDaoContractsReader.{
+  KeyAssigned,
+  KeyState,
+  KeyUnassigned,
+}
+import com.daml.scalautil.Statement.discard
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_BATCH] {
+
+  // Parameters
+
+  private val preparedUpdateLedgerEnd: Connection => PreparedStatement = _.prepareStatement(
+    """
+      |UPDATE
+      |  parameters
+      |SET
+      |  ledger_end = ?,
+      |  ledger_end_sequential_id = ?
+      |
+      |""".stripMargin
+  )
+
+  def updateParams(connection: Connection, params: StorageBackend.Params): Unit = {
+    val preparedStatement = preparedUpdateLedgerEnd(connection)
+    preparedStatement.setString(1, params.ledgerEnd.toHexString)
+    preparedStatement.setLong(2, params.eventSeqId)
+    preparedStatement.execute()
+    preparedStatement.close()
+    ()
+  }
+
+  def ledgerEnd(connection: Connection): StorageBackend.LedgerEnd = {
+    val queryStatement = connection.createStatement()
+    val params = fetch(
+      queryStatement.executeQuery(
+        """
+          |SELECT
+          |  ledger_end,
+          |  ledger_end_sequential_id
+          |FROM
+          |  parameters
+          |
+          |""".stripMargin
+      )
+    )(rs =>
+      StorageBackend.LedgerEnd(
+        lastOffset =
+          if (rs.getString(1) == null) None
+          else Some(Offset.fromHexString(Ref.HexString.assertFromString(rs.getString(1)))),
+        lastEventSeqId = Option(rs.getLong(2)),
+      )
+    )
+    queryStatement.close()
+    assert(params.size == 1)
+    params.head
+  }
+
+  private def fetch[T](resultSet: ResultSet)(parse: ResultSet => T): Vector[T] = {
+    val buffer = mutable.ArrayBuffer.empty[T]
+    while (resultSet.next()) {
+      buffer += parse(resultSet)
+    }
+    resultSet.close()
+    buffer.toVector
+  }
+
+  private val EventSequentialIdBeforeBegin = 0L
+
+  private val TableName: String = "parameters"
```

Comment:
Not necessarily part of this PR, but given the much broader context, it doesn't make sense to have a `TableName` field for `parameters`. I would either inline it or rename it to `ParametersTableName`

Reply:
I placed a todo for parameterstable related refactorings, belongs to that subject.
Interesting you mentioned this, when I started I started with the parameters, did all what you suggested and more houskeping, which was breeding more houskeeping, and eventually I relized if I keep on like this I will never have a chance to finish this PR, neither reviews to review. this is how I arrived to the idea let's keep this manual, let's resist to houskeeping urges(for now)...and thus I throw away all that and copied the whole thing over

Reply:
Makes sense, thanks 👍